plugins/outputs/opentelemetry/opentelemetry_test.go

```diff
@@ -81,6 +86,190 @@ func TestOpenTelemetry(t *testing.T) {
 	require.JSONEq(t, string(expectJSON), string(gotJSON))
 }
 
+func TestOpenTelemetryHTTPProtobuf(t *testing.T) {
+	expect := pmetric.NewMetrics()
+	{
+		rm := expect.ResourceMetrics().AppendEmpty()
+		rm.Resource().Attributes().PutStr("host.name", "potato")
+		rm.Resource().Attributes().PutStr("attr-key", "attr-val")
+		ilm := rm.ScopeMetrics().AppendEmpty()
+		ilm.Scope().SetName("My Library Name")
+		m := ilm.Metrics().AppendEmpty()
+		m.SetName("cpu_temp")
+		m.SetEmptyGauge()
+		dp := m.Gauge().DataPoints().AppendEmpty()
+		dp.Attributes().PutStr("foo", "bar")
+		dp.SetTimestamp(pcommon.Timestamp(1622848686000000000))
+		dp.SetDoubleValue(87.332)
+	}
+
+	var receivedMetrics pmetric.Metrics
+	var receivedContentType string
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		receivedContentType = r.Header.Get("Content-Type")
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			t.Error(err)
+		}
+
+		req := pmetricotlp.NewExportRequest()
+		err = req.UnmarshalProto(body)
+		if err != nil {
+			t.Error(err)
+		}
+
+		receivedMetrics = pmetric.NewMetrics()
+		req.Metrics().CopyTo(receivedMetrics)
+
+		resp := pmetricotlp.NewExportResponse()
+		respBytes, err := resp.MarshalProto()
+		if err != nil {
+			t.Error(err)
+		}
+
+		w.Header().Set("Content-Type", "application/x-protobuf")
+		w.WriteHeader(http.StatusOK)
+		_, err = w.Write(respBytes)
+		if err != nil {
+			t.Error(err)
+		}
+	}))
+	defer server.Close()
+
+	metricsConverter, err := influx2otel.NewLineProtocolToOtelMetrics(common.NoopLogger{})
+	require.NoError(t, err)
+
+	plugin := &OpenTelemetry{
+		ServiceAddress:   server.URL,
+		EncodingType:     "application/x-protobuf",
+		Timeout:          config.Duration(time.Second),
+		Attributes:       map[string]string{"attr-key": "attr-val"},
+		Compression:      "none",
+		metricsConverter: metricsConverter,
+		Log:              testutil.Logger{},
+	}
+	err = plugin.Connect()
+	require.NoError(t, err)
+
+	input := testutil.MustMetric(
+		"cpu_temp",
+		map[string]string{
+			"foo":               "bar",
+			"otel.library.name": "My Library Name",
+			"host.name":         "potato",
+		},
+		map[string]interface{}{
+			"gauge": 87.332,
+		},
+		time.Unix(0, 1622848686000000000))
```

Comment:
Just a matter of taste but I like

```suggestion
		time.Unix(0, 1622848686000000000),
	)
```
to better see the closing bracket.

Reply:
Fixed. https://github.com/influxdata/telegraf/pull/17997/commits/52aa201df03125fe0d3aedcfd05a2ed24146ba44